contest/contest_info.py

```diff
@@ -202,8 +207,40 @@ def write_scoreboard_csv_testcases(writer, contest, scoreboard):
         footer.append(problem.total_solved)
     writer.writerow(footer)
 
-def get_clarifications(user, contest):
+def get_public_user_password_csv(contest):
+    public_contestants = public_user.get_public_contestant(contest)
+    response = HttpResponse(content_type='text/csv')
+    filename = contest.cname.encode('utf-8') + '-password'
+    response['Content-Disposition'] = 'attachment; filename=' + filename
+    
+    #init
+    writer = csv.writer(response)
+    write_public_user_password_csv(writer, contest, public_contestants)
+    
+    return response
 
+def write_public_user_password_csv(writer, contest, public_contestants):
+    header = [contest.cname.encode('utf-8'),str(len(public_contestants))+' users']
+    writer.writerow(header)
+    title = ['Username','Password']
+    writer.writerow(title)
+    for contestant in public_contestants:
+        random_password = get_random_password()
+        user = contestant.user
+        user.password = make_password(random_password)
+        user.save()
+        logger.info('Public user %s changed password' % user.username)
+        user_row = [user.username, random_password]
+        writer.writerow(user_row)
+
+def get_random_password():
```

Comment:
Forget password will lead user to its own profile.
The link will send to its email.